pkg/liqo-controller-manager/namespace-controller/namespace_controller.go

```diff
@@ -0,0 +1,172 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// TODO: put constant value instead of labels
+// TODO: watch also the case of Namespace Creation, now is not managed, important !!!
+
+type NamespaceReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+const (
+	mappingLabel          = "mapping.liqo.io"
+	offloadingLabel       = "offloading.liqo.io"
+	offloadingPrefixLabel = "offloading.liqo.io/"
+)
+
+// TO EVALUATE: when is necessary to update namespace natting table (NNT), if the name exposed in mapping.liqo.io label,
+// is different from the already present, client will change the label value to the old one. For users is not possible
+// to change names of remote namespaces that have already been created with a certain name.
+
+func (r *NamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	ctx := context.Background()
+
+	// TODO: add namespace Finalizer
+	var namespace v1.Namespace
+	if err := r.Get(ctx, req.NamespacedName, &namespace); err != nil {
+		if errors.IsNotFound(err) {
+			klog.Infof("---------------------- I have to delete all entries from NNT, if present")
+			return ctrl.Result{}, nil
+		} else {
+			klog.Error(err, " unable to fetch Namespace for some reasons")
+			return ctrl.Result{}, err
+		}
+	}
+
+	labels := namespace.Labels
```

Comment:
Instead of using this brand-new var, referring to `namespace.Labels` every time would increase code readability

Reply:
Thanks for the advice.